src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestState.cs

```diff
@@ -139,6 +140,16 @@ public void PinReceiveBuffer(byte[] buffer)
             }
         }
 
+        public void DisposeCtrReadFromResponseStream()
+        {
+            if (CtrReadFromResponseStream != null)
+            {
+                WinHttpTraceHelper.Trace("WinHttpRequestState.DisposeCtrReadFromResponseStream: disposing ctr");
+                CtrReadFromResponseStream.Dispose();
+                CtrReadFromResponseStream = default(CancellationTokenRegistration);
```

Comment:
`CancellationTokenRegister` doesn't have any way to determine if it is in a "disposed" state already. But its `Dispose()` method is idempotent. So, I can remove the if() check.